http/main.go

```diff
@@ -741,6 +761,40 @@ func corsMiddleware(next http.Handler) http.Handler {
 	})
 }
 
+func GetNetworkPublicKeyHandler(w http.ResponseWriter, r *http.Request) {
+	fmt.Printf("Got a getNetworkPublicKey request from %s\n", r.RemoteAddr)
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	var req GetNetworkPublicKeyRequest
+	if err := json.Unmarshal(body, &req); err != nil {
+		fmt.Printf("Failed unmarshaling request: %+v body is %+v\n", err, string(body))
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	expectedPk, err := precompiles.GetNetworkPublicKey(int32(req.SecurityZone), &tp)
+	if err != nil {
+		e := fmt.Sprintf("Operation failed: %+v", err)
+		fmt.Println(e)
+		http.Error(w, e, http.StatusBadRequest)
+		return
+	}
+
+	responseData, err := createNetworkPublicKeyResponse(expectedPk)
+	if err != nil {
+		e := fmt.Sprintf("Failed to marshal response: %+v", err)
+		fmt.Println(e)
+		http.Error(w, e, http.StatusInternalServerError)
+		return
```

Comment:
```suggestion
		fmt.Println(err.ErrString()) // don't remember the right syntax here
		http.Error(w, err, http.StatusInternalServerError)
		return
```

Err already contains the same string

Reply:
In this specific case the internal error has the same string, so agree,